[quant] Enable from_float for quantized Embedding_Bag

supriyar opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #43296 [quant] Add benchmarks for quantized embeddingbag module
* **#43176 [quant] Enable from_float for quantized Embedding_Bag**
* #43090 [quant] Make offsets an optional argument
* #43088 [quant] Create nn.quantized.dynamic.EmbeddingBag

Summary:
Convert floating point nn.EmbeddingBag module to
nn.quantized.dynamic.EmbeddingBag module

Test Plan:
python test/test_quantization.py TestDynamicQuantizedModule.test_embedding_bag_api
python test/test_quantization.py TestPostTrainingDynamic.test_embedding_quantization

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23200196](https://our.internmc.facebook.com/intern/diff/D23200196)

supriyar committed:
[quant] Enable from_float for quantized Embedding_Bag
Summary:
Convert floating point nn.EmbeddingBag module to
nn.quantized.dynamic.EmbeddingBag module

Test Plan:
python test/test_quantization.py TestDynamicQuantizedModule.test_embedding_bag_api
python test/test_quantization.py TestPostTrainingDynamic.test_embedding_quantization

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

vkuzo commented on torch/nn/quantized/dynamic/modules/embeddingbag.py:
```diff
@@ -144,4 +145,31 @@ def weight(self):
+        if mod.qconfig is not None and mod.qconfig.weight is not None:
+            weight_observer = mod.qconfig.weight()
+        else:
+            from torch.quantization import PerChannelMinMaxObserver
```
nit: import to top of file?

vkuzo commented on torch/quantization/default_mappings.py:
```diff
@@ -64,7 +64,8 @@
     nn.RNNCell: nnqd.RNNCell,
-    nn.GRUCell: nnqd.GRUCell
+    nn.GRUCell: nnqd.GRUCell,
+    nn.EmbeddingBag: nnqd.EmbeddingBag
```
nit: trailing comma?

supriyar committed:
Update on "[quant] Enable from_float for quantized Embedding_Bag"


Summary:
Convert floating point nn.EmbeddingBag module to
nn.quantized.dynamic.EmbeddingBag module

Test Plan:
python test/test_quantization.py TestDynamicQuantizedModule.test_embedding_bag_api
python test/test_quantization.py TestPostTrainingDynamic.test_embedding_quantization

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

supriyar committed:
Update on "[quant] Enable from_float for quantized Embedding_Bag"


Summary:
Convert floating point nn.EmbeddingBag module to
nn.quantized.dynamic.EmbeddingBag module

Test Plan:
python test/test_quantization.py TestDynamicQuantizedModule.test_embedding_bag_api
python test/test_quantization.py TestPostTrainingDynamic.test_embedding_quantization

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23200196](https://our.internmc.facebook.com/intern/diff/D23200196)

[ghstack-poisoned]

supriyar committed:
Update on "[quant] Enable from_float for quantized Embedding_Bag"


Summary:
Convert floating point nn.EmbeddingBag module to
nn.quantized.dynamic.EmbeddingBag module

Test Plan:
python test/test_quantization.py TestDynamicQuantizedModule.test_embedding_bag_api
python test/test_quantization.py TestPostTrainingDynamic.test_embedding_quantization

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23200196](https://our.internmc.facebook.com/intern/diff/D23200196)

[ghstack-poisoned]
